Floofbot/Services/EventHandlerService.cs

```diff
@@ -491,27 +496,18 @@ public Task UserLeft(IGuildUser user)
                     if (user.IsBot)
                         return;
 
-                    await _userRoleRetentionService.LogUserRoles(user);
 
-                    if ((IsToggled(user.Guild)) == false)
+                    if ((IsToggled(guild)) == false)
                         return;
 
-                    Discord.ITextChannel channel = await GetChannel(user.Guild, "UserLeftChannel");
+                    Discord.ITextChannel channel = await GetChannel(guild, "UserLeftChannel");
                     if (channel == null)
                         return;
 
                     var embed = new EmbedBuilder();
                     embed.WithTitle($"❌ User Left | {user.Username}#{user.Discriminator}")
                          .WithColor(Color.Red)
                          .WithDescription($"{user.Mention} | ``{user.Id}``");
-                    if (user.JoinedAt != null)
```

Comment:
So what happened to `JoinedAt`? I can't seem to find older docs on previous versions of Discord.NET, and there is no entry for `JoinedAt` for version 3.

Reply:
Because it's changed from guilduser to socketuser, this no longer knows the join date as the user isn't associated with a guild, couldn't think of alternative solution but to remove it

Reply:
This is really late, but you can actually get the `SocketGuildUser` by using [`GetUser`](https://discordnet.dev/api/Discord.WebSocket.SocketGuild.html#Discord_WebSocket_SocketGuild_GetUser_System_UInt64_).